src/wallet/api/swapService.ts

```diff
@@ -172,3 +173,70 @@ export const get = async (proposalId: string, password: string): Promise<AtomicS
     throw err;
   }
 };
+
+interface SwapUpdateParams {
+  proposalId: string;
+  password: string;
+  partialTx: string;
+  version: number;
+  signatures?: string;
+}
+
+/**
+ * Updates the proposal on the Atomic Swap Service with the parameters informed
+ */
+export const update = async (params: SwapUpdateParams):
+  Promise<{ success: boolean }> => {
+  validateParameters();
+  const { proposalId, password, partialTx, version, signatures } = params;
+
+  const swapAxios = await axiosInstance();
+  const options = {
+    headers: { 'X-Auth-Password': hashPassword(password) }
+  } as AxiosRequestConfig;
+
+  const payload = {
+    partialTx: encryptString(partialTx, password),
+    version,
+    signatures: signatures ? encryptString(signatures, password) : null,
```

Comment:
💬 The signature will be validated in the server, right? By validating the signature in the lib we can avoid a round trip to the server. Does it worth the effort to validate it here?

Reply:
The server does not validate the signatures: it only accepts whatever value is passed and stores it on the database ( [code reference](https://github.com/HathorNetwork/hathor-atomic-swap-service/pull/4/files#diff-a7489a11dd17a7d3af1075dbca6440d8c18a882f6ec1f077991e2b2f1a8e1478) ).

If the value is `null`, it considers the signatures should be voided and also updates the database's column with `null`.

Reply:
Got it!